src/Orleans/Utils/FastLock.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Orleans.Runtime
+{
+    // About 2x faster than ordinary lock on mostly single threaded locking
+    internal class FastLock
```

Comment:
might want to consolidate with `AsyncSerialExecutor.InterlockedExchangeLock` to avoid having 2 similar implementations